modules/swift/VectorSearchSnippets.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  VectorSearchSnippets.swift
+//
+//  Created by Vlad Velicu on 20/03/2024.
+//
+
+import Foundation
+import CouchbaseLiteSwift
+import NaturalLanguage
+
+class VectorSearchSnippets {
+    var database: Database!
+    var collection: Collection!
+    var model: NLEmbedding!
+
+    // MARK: Configuring a project to use Vector Search.
+
+    /*/
+     Download Couchbase Lite iOS Swift and Couchbase Lite Vector Search libraries
+     Extract the xcframeworks and copy them into the App Project Directory
+     
+     Add both libraries to the *Frameworks, Libraries and Embedded Content* of your desired target
+     */
+
+    func vectorIndex() throws {
+        database = try Database(name: "vectorDB")
+        collection = try database.defaultCollection()
+        
+        // MARK: Create a default Vector Index Configuration
+        
+        var config = VectorIndexConfiguration(expression: "string", dimensions: 300, centroids: 20)
+        
+        // Default values already set:
+        print(config.encoding) // scalarQuantizer(type: .SQ8)
+        print(config.metric) // euclidean
+        print(config.minTrainingSize) // 25 * centroids(20)
+        print(config.maxTrainingSize) // 256 * centroids(20)
+        
+        // MARK: Set custom optional settings
+        config.encoding = .none
+        config.metric = .cosine
+        config.minTrainingSize = 50
+        config.maxTrainingSize = 300
+        
+    }
+
+    func vectorIndexEmbedding() throws -> ResultSet? {
+        // MARK: Create Vector Index with Embedding
+        let config = VectorIndexConfiguration(expression: "word", dimensions: 300, centroids: 20)
+        try collection.createIndex(withName: "vector_index", config: config)
+        
+        model = NLEmbedding.wordEmbedding(for: .english)!
```

Comment:
The code from this line is not relavent. There is also `useVectorMatch()` that shows the query.